File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/domain/pullrequests/PullRequestDomainContract.java
package com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests;

import com.duxtinto.intellij.plugin.github.codereviews.net.pullrequests.PullRequestQueryParameters;

import javax.annotation.Nonnull;
import java.io.IOException;
import java.util.List;

public interface PullRequestDomainContract {
    interface Fetcher {
        List<PullRequestEntity> fetchForRepository(@Nonnull String userName, @Nonnull String repoName, PullRequestQueryParameters parameters) throws IOException;
    }

    interface Repository {
        List<PullRequestEntity> getAllOpenBy(@Nonnull String userName, @Nonnull String repoName) throws IOException;
    }

    interface DescriptionParser {
        PullRequestDescriptionEntity parse(String description);
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/domain/pullrequests/PullRequestDomainContract.kt
package com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests

import com.duxtinto.intellij.plugin.github.codereviews.net.pullrequests.apiv3.PullRequestQueryParameters

interface PullRequestDomainContract {
    interface Fetcher {
        fun fetchAllForRepository(userName: String, repoName: String, parameters: PullRequestQueryParameters): List<PullRequestEntity>
        fun fetchOneForRepository(userName: String, repoName: String, id: Long): PullRequestEntity
    }

    interface Repository {
        fun getAllMyAssignedBy(userName: String, repoName: String): List<PullRequestEntity>
        fun getOneById(userName: String, repoName: String, id: Long): PullRequestEntity?
        fun getAllMyRequestedBy(userName: String, repoName: String): List<PullRequestEntity>
    }

    interface DescriptionParser {
        fun parse(description: String): PullRequestDescriptionEntity
    }
}
